liana-gui/Cargo.toml

```diff
@@ -17,7 +17,7 @@ path = "src/main.rs"
 async-trait = "0.1"
 async-hwi = { version = "0.0.24" }
 liana = { git = "https://github.com/wizardsardine/liana", branch = "master", default-features = false, features = ["nonblocking_shutdown"] }
```

Comment:
maybe we should point to the local repo?

```suggestion
liana = { path = "../liana", default-features = false, features = ["nonblocking_shutdown"] }
```

but doing so we fail building:
<details>

```shell
❯ cargo build --release                                                                                                                 
   Compiling liana v8.0.0 (/home/pyth/rust/liana/liana)
   Compiling liana-gui v8.0.0 (/home/pyth/rust/liana/liana-gui)
error[E0061]: this function takes 2 arguments but 1 argument was supplied
   --> liana-gui/src/daemon/embedded.rs:25:22
    |
25  |         let handle = DaemonHandle::start_default(config.clone()).map_err(DaemonError::Start)?;
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^---------------- an argument of type `bool` is missing
    |
note: associated function defined here
   --> /home/pyth/rust/liana/liana/src/lib.rs:565:12
    |
565 |     pub fn start_default(
    |            ^^^^^^^^^^^^^
help: provide the argument
    |
25  |         let handle = DaemonHandle::start_default(config.clone(), /* bool */).map_err(DaemonError::Start)?;
    |                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0061]: this function takes 2 arguments but 1 argument was supplied
   --> liana-gui/src/installer/mod.rs:346:25
    |
346 |     let daemon_handle = liana::DaemonHandle::start_default(cfg);
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^----- an argument of type `bool` is missing
    |
note: associated function defined here
   --> /home/pyth/rust/liana/liana/src/lib.rs:565:12
    |
565 |     pub fn start_default(
    |            ^^^^^^^^^^^^^
help: provide the argument
    |
346 |     let daemon_handle = liana::DaemonHandle::start_default(cfg, /* bool */);
    |                                                           ~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0061`.
error: could not compile `liana-gui` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...

```

</details>

and should apply this kind of patch but i don't really understand why it don't fail otherwise:
<details>

```diff
diff --git a/liana-gui/src/daemon/embedded.rs b/liana-gui/src/daemon/embedded.rs
index c4f77d7..17e1e3a 100644
--- a/liana-gui/src/daemon/embedded.rs
+++ b/liana-gui/src/daemon/embedded.rs
@@ -18,6 +18,10 @@ pub struct EmbeddedDaemon {
 
 impl EmbeddedDaemon {
     pub fn start(config: Config) -> Result<EmbeddedDaemon, DaemonError> {
+        #[cfg(not(all(unix, feature = "daemon")))]
+        let handle =
+            DaemonHandle::start_default(config.clone(), true).map_err(DaemonError::Start)?;
+        #[cfg(all(unix, feature = "daemon"))]
         let handle = DaemonHandle::start_default(config.clone()).map_err(DaemonError::Start)?;
         Ok(Self {
             handle: Mutex::new(Some(handle)),
@@ -31,6 +35,7 @@ impl EmbeddedDaemon {
     {
         match self.handle.lock().await.as_mut() {
             Some(DaemonHandle::Controller { control, .. }) => method(control),
+            Some(DaemonHandle::Server { .. }) => unimplemented!(),
             None => Err(DaemonError::DaemonStopped),
         }
     }
diff --git a/liana-gui/src/installer/mod.rs b/liana-gui/src/installer/mod.rs
index e0adac8..fefe7ce 100644
--- a/liana-gui/src/installer/mod.rs
+++ b/liana-gui/src/installer/mod.rs
@@ -334,7 +334,11 @@ impl Installer {
 
 pub fn daemon_check(cfg: liana::config::Config) -> Result<(), Error> {
     // Start Daemon to check correctness of installation
-    match liana::DaemonHandle::start_default(cfg) {
+    #[cfg(not(all(unix, feature = "daemon")))]
+    let daemon_handle = liana::DaemonHandle::start_default(cfg, true);
+    #[cfg(all(unix, feature = "daemon"))]
+    let daemon_handle = liana::DaemonHandle::start_default(cfg);
+    match daemon_handle {
         Ok(daemon) => daemon
             .stop()
             .map_err(|e| Error::Unexpected(format!("Failed to stop Liana daemon: {}", e))),

```
</details>

Reply:
it looks to me it's related to changes in 191249a871922e2fbdf8697d7a733c572a9a08ad

Reply:
I think this change is sadly out of scope for  now, we need to clean the liana crate and remove the `daemon`feature that is not good in my opinion.

Reply:
i agree, services should be managed by systemd